### PR TITLE
[TAS-1093] ✨ Add coupon section in book and collection

### PR DESCRIPTION
**Pull request by williamchong**
Requires: https://github.com/likecoin/likecoin-api-public/pull/618
image
image

TODO: refactor ui and shared function

### CODE REVIEW

File: pages/nft-book-store/collection/status/[collectionId].vue [hunk 1]
```diff
@@ -226,6 +226,64 @@
...
+          label="New coupon code"
+          :ui="{ label: { base: 'font-mono font-bold' } }"
+        >
+          <UInput
+            v-model="newCoupon.id"
+            placeholder="coupon_code"
+          />
+        </UFormGroup>
+        <UFormGroup
+          label="Coupon discount multiplier, 0.01x - 1x"
```
[the remark this answers is not in the document]
Comment (williamchong) [reply pages/nft-book-store/collection/status/[collectionId].vue:260]: yes, 1x = no discount

[the remark this answers is not in the document]
Comment (nwingt) [reply pages/nft-book-store/collection/status/[collectionId].vue:260]: In what scenario is there no discount? Why don't we offer `0x` for free?

[the remark this answers is not in the document]
Comment (williamchong) [reply pages/nft-book-store/collection/status/[collectionId].vue:260]: No scenario actually, but i have no idea what should be "minimal discount"
A free code would currently break the purchase flow (stripe does not supporr free checkout), but we can make it work if needed with extra effort

[the remark this answers is not in the document]
Comment (nwingt) [reply pages/nft-book-store/collection/status/[collectionId].vue:260]: I'll open a task in the backlog about supporting a 100% discount coupon then

**Pull request merged**